src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs

```diff
@@ -0,0 +1,108 @@
+module FSharpLint.Rules.NoPartialFunctions
+
+open System
+open FSharp.Compiler.Range
+open FSharpLint.Framework
+open FSharpLint.Framework.Suggestion
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.Rules
+
+[<RequireQualifiedAccess>]
+type Config = {
+    AllowedPartials:string list
+    AdditionalPartials:string list
+}
+
+type private Replacement =
+    | PatternMatch
+    | Function of functionName:string
+
+let private partialFunctionIdentifiers =
```

Comment:
@Krzysztof-Cieslak I added all the partial collection functions that I'm aware of, anything else we should add?

Reply:
@jgardella how about Map.Item -> Map.TryFind

Reply:
@knocte At the moment it'll require a little bit more work to handle methods on an object instance (some type checking) - might be worth another PR for that, we could potentially add:

`Map.find` -> `Map.tryFind`
`Map.findKey` -> `Map.tryFindKey`